File: src/game/powerups/ArmorPowerUp.java
package game.powerups;

import game.entity.models.BomberEntity;
import game.entity.models.Coordinates;
import game.utils.Paths;

import java.awt.image.BufferedImage;

public class ArmorPowerUp extends PowerUp {
    /**
     * Constructs an entity with the given coordinates.
     *
     * @param coordinates the coordinates of the entity
     */
    public ArmorPowerUp(Coordinates coordinates) {
        super(coordinates);
    }

    @Override
    public BufferedImage getImage() {
        return loadAndSetImage(Paths.INSTANCE.getPowerUpsFolder() + "/armor_up.png");
    }

    @Override
    public int getDuration() {
        return 15;
    }

    @Override
    protected void doApply(BomberEntity entity) {
        entity.setImmune(true);
    }

    @Override
    protected void cancel(BomberEntity entity) {
        if(entity.isSpawned())
            entity.setImmune(false);
    }
}

File: src/game/powerups/ArmorPowerUp.kt
package game.powerups

import game.entity.EntityTypes
import game.entity.player.BomberEntity
import game.entity.models.Coordinates
import game.utils.Paths.powerUpsFolder
import java.awt.image.BufferedImage

class ArmorPowerUp : PowerUp {
    constructor(id: Long) : super(id)
    constructor(coordinates: Coordinates?) : super(coordinates)

    override fun getImage(): BufferedImage = loadAndSetImage("$powerUpsFolder/armor_up.png")

    override val duration: Int
        get() = DEFAULT_DURATION_SEC

    override fun doApply(entity: BomberEntity) {
        entity.isImmune = true
    }

    override fun cancel(entity: BomberEntity) {
        if (entity.isSpawned) entity.isImmune = false
    }

    override val type: EntityTypes
        get() = EntityTypes.ArmorPowerUp
}